### DISCUSSION TITLE
replace regexp for additionalBranchPrefix

### DISCUSSION THREAD

**Discussion by pebo** [renovatebot/renovate#20729] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

How can we add a `/` after `packageFileDir` in `additionalBranchPrefix` but skip that for files in the repository root dir (`/`) to avoid getting `//` as a prefix?

The following expression does not work:
```json
  "additionalBranchPrefix": "{{{ replace '(.*)\\/$' '$1' packageFileDir }}}/",
```

The root folder contains the file `/docker-compose.yml` and the error we get is:

````
 "message": "fatal: 'renovate//prom-prometheus-2.x' is not a valid branch name\n",
````

What's the correct syntax for dropping a (trailing) slash in `additionalBranchPrefix`?

### What you tried so far.

```json
  "additionalBranchPrefix": "{{{ replace '(.*)\\/$' '$1' packageFileDir }}}/",
```

### Relevant debug logs

<details><summary>Logs</summary>

```
WARN: Error updating branch(branch="renovate//prom-prometheus-2.x")
{
  "err": {
    "task": {
      "commands": [
        "checkout",
        "-B",
        "renovate//prom-prometheus-2.x",
        "origin/master"
      ],
      "format": "utf-8",
      "parser": "[function]"
    },
    "message": "fatal: 'renovate//prom-prometheus-2.x' is not a valid branch name\n",```

</details>

**Comment by RahulGautamSingh** [comment, marked as answer]
You can try this:
`"additionalBranchPrefix": "{{#if packageFileDir}}{{packageFileDir}}/{{/if}}"`

Reproduced it [here](https://github.com/RahulGautamSingh-testing/additionalBranchPrefix/pulls)

**Reply by pebo** [reply]
Thank you! This solved the issue - `packageFileDir` seems to be the empty string for the root folder.
